added the importance: required|recommended|optional to each of the fi…

virag-compbio opened the pull request:
Added the "importance" field in the markdown files corresponding to the individual sheets in the ghga_submission_full.xlsx

Understandably, the importance field could take one of three values:
1. required
2. recommended
3. optional

As of now, there is only one field "required" which is a boolean (True|False), which is also not so intuitive for experimentalists/people working on the bench.

sbilge reviewed (changes requested):
Hello Virag, I noticed that the script does not appear to have been updated to reflect the changes you intended for the `data_dictionary` files. These files are autogenerated by the `update_metadata_docs.py` script located in the scripts folder, which should have been modified accordingly.

virag-compbio commented:
> Hello Virag, I noticed that the script does not appear to have been updated to reflect the changes you intended for the `data_dictionary` files. These files are autogenerated by the `update_metadata_docs.py` script located in the scripts folder, which should have been modified accordingly.

Hi Bilge,
The update_metatdata_docs.py has been updated. And the "significance" slot is also added to the template file i.e. docs/templates/sheet_documentation_template.md.jinja

Please note that I have renamed the slot "importance" to "significance" now. Everything else remains the same.

Thanks

mauerk commented:
Hi Virag, thanks for adding this missing information. My idea would be to deprecate the "required: True / False" flag in the documentation and change the "significance" to "requirement", which can take one of "optional / recommended / required" to avoid redundancy. As an visual aid, "required" could be highlighted with a different text color (if possible). What do you think?

sbilge commented on scripts/update_metadata_docs.py:
```diff
@@ -142,6 +142,11 @@ def extract_slots_from(
         [SchemaView, Union[str, None]], Union[dict, None]
     ] = extract_permissible_values,
 ) -> list[dict]:
+
```
There is a doc string between function types and a redundant looping. Thus, the dynamic attribute assignment is not needed. 
```suggestion
    """Extract slot information of a given class. Additionally add the slot significance which takes one the 3 values: required, recommended and optional.
    """

    return [
        {
            "name": slot.name,
            "description": slot.description,
            "data_type": {
                "range": slot.range,
                "enum": add_enum(schema, slot.range),
            },
            "required": slot.required,
            "significance": "required" if slot.required else "recommended" if slot.recommended else "optional"
        }
        for slot in schema.class_induced_slots(sheet_name)
    ]
```

virag-compbio commented:
> Hi Virag, thanks for adding this missing information. My idea would be to deprecate the "required: True / False" flag in the documentation and change the "significance" to "requirement", which can take one of "optional / recommended / required" to avoid redundancy. As an visual aid, "required" could be highlighted with a different text color (if possible). What do you think?

Hi Karo,
Thanks for the suggestion regarding the coloring scheme.
I have implemented that now and the required field is removed.

So we now have the field/slot called requirement which can take one of the three values:
1. Required - colored in red
2. Recommended - colored in blue
3. Optional - colored in green

The template file has been modified accordingly.
